src/commands/api-mesh/log-get-bulk.js

```diff
@@ -0,0 +1,208 @@
+		// Regular expression to validate the input date format YYYY-MM-DDTHH:MM:SSZ
+		const dateTimeRegex = /^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}Z$/;
+
+		// Validate startTime format
+		if (!dateTimeRegex.test(flags.startTime)) {
+			this.error('Invalid startTime format. Please use the format YYYY-MM-DDTHH:MM:SSZ');
+			return;
+		}
+
+		// Validate endTime format
+		if (!dateTimeRegex.test(flags.endTime)) {
+			this.error('Invalid endTime format. Please use the format YYYY-MM-DDTHH:MM:SSZ');
+			return;
+		}
+
+		// Properly format startTime and endTime strings before handing it over to SMS
+		const formattedStartTime = flags.startTime.replace(/-|:|Z/g, '').replace('T', 'T');
+		const formattedEndTime = flags.endTime.replace(/-|:|Z/g, '').replace('T', 'T');
+
+		// Convert formatted times to Date objects for comparison
+		const startTime = new Date(flags.startTime);
+		const endTime = new Date(flags.endTime);
+
+		// Require both startTime and endTime
+		if (!startTime || !endTime) {
+			this.error('Provide both startTime and endTime');
+			return;
+		}
+
+		// Get the current date and calculate the date 30 days ago, both in UTC
+		const today = new Date();
+		const thirtyDaysAgo = new Date(today);
+		thirtyDaysAgo.setUTCDate(today.getUTCDate() - 30);
+		// Validate that logs from beyond 30 days from today are not available
+		if (startTime < thirtyDaysAgo || endTime < thirtyDaysAgo) {
+			this.error('Cannot get logs older than 30 days from today. Please adjust your time range.');
+			return;
+		}
+
+		// Validate filepath
+		if (!filename) {
+			this.error(
+				'Missing file path. Please provide a valid file in the current working directory.',
+			);
+			return;
+		}
+
+		// Check if the file exists
+		const outputFile = path.resolve(process.cwd(), filename);
+
+		// Ensure file exists and is empty before proceeding
+		if (!fs.existsSync(outputFile)) {
+			throw new Error(`Specified file doesn't exist in the ${process.cwd()}`);
+		}
+
+		const stats = fs.statSync(outputFile);
+		if (stats.size > 0) {
+			throw new Error(`Please make sure that file: ${filename} is empty`);
+		}
+		// truncate milliseconds to ensure comparison is only done up to seconds
+		startTime.setMilliseconds(0);
+		endTime.setMilliseconds(0);
+
+		// Validate startTime < endTime
+		if (startTime > endTime) {
+			this.error('endTime should be greater than startTime');
+		}
+
+		// 4. Check if the duration between start and end times is greater than 30 minutes (1800 seconds)
+		const timeDifferenceInSeconds = (endTime.getTime() - startTime.getTime()) / 1000;
+
+		if (timeDifferenceInSeconds > 1800) {
+			const hours = Math.floor(timeDifferenceInSeconds / 3600); //hours calculation
+			const minutes = Math.floor((timeDifferenceInSeconds % 3600) / 60); //minutes calculation
+			const seconds = timeDifferenceInSeconds % 60; //seconds calculation
+
+			this.error(
+				`Max duration between startTime and endTime should be 30 minutes. Current duration is ${hours} hour${
```

Comment:
Add a period after the second sentence

Reply:
@jhadobe. The period is added on line 112.
This is how it will be output to the user
```
Max duration between startTime and endTime should be 30 minutes. Current duration is 1 hour 10 
minutes and 45 seconds.
```